katdal/test/test_datasources.py

```diff
@@ -98,28 +103,45 @@ def make_fake_datasource(telstate, store, cbid, l0_shape, l1_flags_shape=None,
     return view_l0_capture_stream(telstate, cbid, 'sdp_l0') + (l0_data, l1_flags_data)
 
 
+def assert_telstate_data_source_equal(source1, source2):
+    """Assert that two :class:`~katdal.datasources.TelstateDataSource`s are equal."""
+    # Check attributes (sensors left out for now)
+    keys = source1.telstate.keys()
+    assert set(source2.telstate.keys()) == set(keys)
+    for key in keys:
+        np.testing.assert_array_equal(source1.telstate[key], source2.telstate[key])
+    # Check that we also have the same telstate view in both sources
+    assert source1.telstate.prefixes == source2.telstate.prefixes
+    # Check data arrays
+    np.testing.assert_array_equal(source1.timestamps, source2.timestamps)
+    np.testing.assert_array_equal(source1.data.vis.compute(), source2.data.vis.compute())
+    np.testing.assert_array_equal(source1.data.flags.compute(), source2.data.flags.compute())
+    np.testing.assert_array_equal(source1.data.weights.compute(), source2.data.weights.compute())
+
+
 class TestTelstateDataSource(object):
     def setup(self):
         self.tempdir = tempfile.mkdtemp()
         self.store = NpyFileChunkStore(self.tempdir)
         self.telstate = katsdptelstate.TelescopeState()
-        self.cbid = 'cb'
 
     def teardown(self):
         shutil.rmtree(self.tempdir)
 
-    def test_timestamps(self):
-        view, cbid, sn, l0_data, l1_flags_data = \
-            make_fake_datasource(self.telstate, self.store, self.cbid, (20, 64, 40))
-        data_source = TelstateDataSource(view, cbid, sn, self.store)
-        np.testing.assert_array_equal(
-            data_source.timestamps,
-            np.arange(20, dtype=np.float32) * 2 + 123456912)
+    def test_basic_timestamps(self):
+        # Add a sensor to telstate to exercise the relevant code paths in TelstateDataSource
+        self.telstate.add('obs_script_log', 'Digitisers synced', ts=123456789., immutable=False)
```

Comment:
```suggestion
        self.telstate.add('obs_script_log', 'Digitisers synced', ts=123456789.)
```
Some day I'd like to deprecate the `immutable` parameter, since there is `__setitem__` for immutables.